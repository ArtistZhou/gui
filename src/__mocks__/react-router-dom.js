import React from 'react';
import { compose, setDisplayName } from 'recompose';
import * as rrd from 'react-router-dom';
// Just render plain div with its children
rrd.BrowserRouter = compose(setDisplayName('BrowserRouter'))(({ children }) => <div>{children}</div>);

module.exports = rrd;
