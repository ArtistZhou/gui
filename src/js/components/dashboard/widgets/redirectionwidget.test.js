import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';
import RedirectionWidget from './redirectionwidget';

describe('RedirectionWidget Component', () => {
  it('renders correctly', () => {
    const tree = createMount()(
      <MemoryRouter>
        <RedirectionWidget target="testlocation" buttonContent={<div />} />
      </MemoryRouter>
    );
    expect(tree.html()).toMatchSnapshot();
  });
});
