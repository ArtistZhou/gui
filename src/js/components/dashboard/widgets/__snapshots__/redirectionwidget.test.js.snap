// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="onboard widget"
  onClick={[Function]}
>
  <div>
    <p
      className="muted"
    />
  </div>
  <a
    className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-text-3 MuiButton-flat-6"
    href="/testlocation"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex="0"
  >
    <span
      className="MuiButton-label-2"
    >
      <div />
    </span>
    <span
      className="MuiTouchRipple-root-30"
    />
  </a>
</div>
`;
